### PR TITLE
Modal | Custom popup

### LINKED ISSUE #182: Modal | Custom Popup
To have the custom popup inside the modal.

Currently, the modal has a default title, body, and footer. It should be able to pass the active view property for dynamic content.

image

### LINKED ISSUE #183: Modal | Clicking on background to dismiss modal should be disabled
Clicking on background to dismiss modal should be disabled

### LINKED ISSUE #184: Popup
The popup should have the property close when clicked outside and a custom UI inside the popup.

### PR DESCRIPTION
**Pull request by BAL4-R**
Description: 
1. Enable User to create custom popup, closes #182 
2. Background click is disabled for modal, closes #183 
3. Closes #184 

Screenshot:
image
Updated ImageButtons with TMButton
image

### CODE REVIEW

File: DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <ImageButton Clicked="ArrowDownClicked"
+                 Source="{x:Static constants:ImageConstants.ArrowDown}"
+                 AbsoluteLayout.LayoutBounds="50,50" />
+    <ImageButton Clicked="ArrowLeftClicked"
+                 Source="{x:Static constants:ImageConstants.ArrowLeft}"
+                 AbsoluteLayout.LayoutBounds="300,50" />
+    <ImageButton Clicked="ArrowRightClicked"
+                 Source="{x:Static constants:ImageConstants.ArrowRight}"
+                 AbsoluteLayout.LayoutBounds="50,650" />
+    <ImageButton Clicked="ArrowUpClicked"
```
Comment (Sofiya-kumar) [inline DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml:18]: Can we have TMButton here for all the image buttons?

**Review by Sofiya-kumar — APPROVED** [review]

File: DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml (hunk 1 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml]: Done

**Review by ElishaSamPeterPrabhu — APPROVED** [review]

**Review by rthanga1 — APPROVED** [review]

**Review by rthanga1 — DISMISSED** [review]

File: DemoApp/DemoApp/Constant/ImageConstants.cs [hunk 2]
```diff
@@ -13,5 +13,9 @@ internal static class ImageConstants
         public const string PasswordIcon = "password_eye.png";
```
Comment (rthanga1) [inline DemoApp/DemoApp/Constant/ImageConstants.cs:13]: Update closes #ticketNumber in the PR

Comment (BAL4-R) [reply DemoApp/DemoApp/Constant/ImageConstants.cs:13]: I added #184 in the description. Is that the requirement?

Comment (rthanga1) [reply DemoApp/DemoApp/Constant/ImageConstants.cs:13]: Yeah, We should have a 'ShouldCloseWhenTappedOutside' property.

Comment (BAL4-R) [reply DemoApp/DemoApp/Constant/ImageConstants.cs:13]: There is already a property `CloseWhenBackgroundIsClicked` in the PopupPage.cs (used in `SampleCustomPopup.xaml` line 9).

For TMModal, The TMModal.Show() method takes that as an argument, with default value false.

Comment (rthanga1) [reply DemoApp/DemoApp/Constant/ImageConstants.cs:13]: Then, it is fine.

File: DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:constants="clr-namespace:DemoApp.Constant"
+             xmlns:Modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
+             x:Class="DemoApp.Views.SamplePopupPage"
+             Title="SamplePopupPage">
+  <AbsoluteLayout>
+    <Modus:TMButton Clicked="ArrowDownClicked"
```
Comment (Sofiya-kumar) [inline DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml:9]: Can you please try with click event with Viewmodel

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Views/PopupSamples/SampleCustomPopup.xaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<popup:PopupPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
```
Comment (Sofiya-kumar) [inline DemoApp/DemoApp/Views/PopupSamples/SampleCustomPopup.xaml:2]: Popup should be from modus components please check the namespace

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Views/PopupSamples/SampleCustomPopup.xaml (hunk 4 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/Views/PopupSamples/SampleCustomPopup.xaml]: Done

File: DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml (hunk 3 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/Views/PopupSamples/SamplePopupPage.xaml]: Done

**Review by rthanga1 — APPROVED** [review]

**Review by Sofiya-kumar — APPROVED** [review]

**Pull request merged**